### PR TITLE
Add trigonometric functions for special irrationals

### LINKED ISSUE #20: Exact results for triangular functions
In the current master, the return value of triangular function is not exact.
```julia
julia> using Tau

julia> sin(τ)
-2.4492935982947064e-16
```
This can be solved like https://github.com/JuliaLang/julia/pull/42595.

### PR DESCRIPTION
**Pull request by devmotion**
Fixes #20.

This PR is a (largely) reduced and updated version of #14. Arguably it is quite uncontroversial since it just matches the existing behaviour of `sin(pi)` etc. in Base: https://github.com/JuliaLang/julia/pull/42595

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -150,3 +150,28 @@ end
...
+    # Check consistency of definitions
+    @test sincos(x) === (sin(x), cos(x))
+    @test tan(x) === sin(x) / cos(x)
+
+    # These are defined automatically via sin, cos, and tan
+    @test csc(x) === inv(sin(x))
+    @test sec(x) === inv(cos(x))
+    @test cot(x) === inv(tan(x))
+  end
+end
```
Comment (hyrodium) [inline test/runtests.jl]: These tests should have indents with four spaces.

**Review by hyrodium — COMMENTED** [review]
Note that `sec(quartπ)` does not have consistent accuracy because `sec(x)` is defined by `inv(cos(x))`.
```julia
julia> sec(quartπ)
1.414213562373095

julia> Float64(sqrt2)
1.4142135623730951

julia> inv(invsqrt2)  # same as inv(cos(invsqrt2))
1.414213562373095
```
This is one of the reasons why I would like to have `sec(quartπ) === sqrt2` in #14.

**Comment by devmotion** [comment]
> This is one of the reasons why I would like to have sec(quartπ) === sqrt2 in https://github.com/JuliaMath/IrrationalConstants.jl/pull/14.

I think we have to be consistent and always return `Float64` (consistent with Base), otherwise it gets completely confusing. I improved the accuracy slightly by defining `sec` and `csc` for `quartπ`.

**Comment by hyrodium** [comment]
>I think we have to be consistent and always return `Float64` (consistent with Base),

There is a counter example `log(ℯ) === 1` (https://github.com/JuliaMath/IrrationalConstants.jl/pull/14#issuecomment-1107851681).

>otherwise it gets completely confusing.

I thought "return Irrational if we have the irrational, otherwise, return Float64" is the best approach (https://github.com/JuliaMath/IrrationalConstants.jl/pull/14#issuecomment-1108226555), and I could not find that much confusion here.

However, I agree with merging this PR because this PR just improves the accuracy and there are no breaking changes. My last question is why my PR #14 is not merged yet. I updated the PR to return `Float64` for all `Irrational` inputs as you suggested.

**Comment by devmotion** [comment]
> There is a counter example

The `log` definition is much older than the trigonometric definitions in Base and the odd one, it seems. I think when defining trigonometric functions for irrationals we should be consistent with how in Base trigonometric funtions for irrationals are implemented. And the reasoning there was quite clear, they should all return `Float64`.

> My last question is why my PR https://github.com/JuliaMath/IrrationalConstants.jl/pull/14 is not merged yet. I updated the PR to return Float64 for all Irrational inputs as you suggested.

It seems unfortunately I lost track of it, and with the recent changes it became outdated. Hence I decided to open a new PR. Generally, I also prefer smaller PRs with fewer changes (e.g., focused only on trigonometric functions) and without additional formatting changes.

**Pull request merged**